Source/DafnyCore/Verifier/BoogieGenerator.Methods.cs

```diff
@@ -168,7 +168,9 @@ void AddFunction_Top(Function f, bool includeAllMethods) {
     }
 
     void AddMethod_Top(MethodOrConstructor m, bool isByMethod, bool includeAllMethods) {
-      if (!includeAllMethods && !InVerificationScope(m) && !referencedMembers.Contains(m)) {
+      if (!includeAllMethods &&
+          !m.EnclosingClass.EnclosingModuleDefinition.EnclosingLiteralModuleDecl.ShouldVerify(program.Compilation) &&
```

Comment:
I still don't know what `InVerificationScope` means so this predicate looks more meaningful.